python/cudf/bindings/parquet.pyx

```diff
@@ -37,76 +33,60 @@ cpdef cpp_read_parquet(path, columns=None, row_group=None, skip_rows=None,
     cudf.io.parquet.to_parquet
     """
 
-    # Setup arguments
-    cdef pq_read_arg pq_reader = pq_read_arg()
+    # Setup reader options
+    cdef ParquetReaderOptions options = ParquetReaderOptions()
+    for col in columns or []:
+        options.columns.push_back(str(col).encode())
+    options.strings_to_categorical = strings_to_categorical
 
+    # Create reader from source
     if not os.path.isfile(path) or not os.path.exists(path):
         raise FileNotFoundError(
             errno.ENOENT, os.strerror(errno.ENOENT), path
         )
-    path = str(path).encode()
-    source_ptr = <char*>path
-    pq_reader.source_type = FILE_PATH
-    pq_reader.source = source_ptr
-
-    usecols = columns
-    arr_cols = []
-    pq_reader.use_cols_len = 0
-    cdef vector[const char*] vector_use_cols
-    if usecols is not None:
-        for col in usecols:
-            arr_cols.append(str(col).encode())
-        pq_reader.use_cols_len = len(usecols)
-
-    vector_use_cols = arr_cols
-    pq_reader.use_cols = vector_use_cols.data()
-
-    if row_group is not None:
-        pq_reader.row_group = row_group
-    else:
-        pq_reader.row_group = -1
+    cdef unique_ptr[ParquetReader] reader
+    reader = unique_ptr[ParquetReader](
+        new ParquetReader(str(path).encode(), options)
+    )
 
+    # Read data into columns
+    cdef cudf_table table
```

Comment:
Do we need to free this or is it `cudf::table` stack allocated in libcudf?

Reply:
It's stack allocated in libcudf, so no need to free...